fs/backgroundfetcher/background_fetcher.go

```diff
@@ -168,3 +179,16 @@ func (bf *BackgroundFetcher) Run(ctx context.Context) error {
 		}
 	}
 }
+
+func (bf *BackgroundFetcher) emitMetric() {
+	for {
+		select {
```

Comment:
similar to `bf.Run()` we should pass the `ctx` to `emitMetric` and have a `case` for `ctx.Done()`:

```
		case <-ctx.Done():
			return
```

Reply:
Done